src/Storage/Controllers/InstancesController.cs

```diff
@@ -155,7 +161,22 @@ public class InstancesController : ControllerBase
             {
                 if (instanceOwnerPartyId == null)
                 {
-                    return BadRequest("InstanceOwnerPartyId must be defined.");
+                    if (string.IsNullOrEmpty(instanceOwnerIdentifier))
+                    {
+                        return BadRequest("Either InstanceOwnerPartyId or InstanceOwnerIdentifier must need to be defined.");
+                    }
+
+                    (string instanceOwnerIdType, string instanceOwnerIdValue) = InstanceHelper.GetIdentifierFromInstanceOwnerIdentifier(instanceOwnerIdentifier);
+
+                    if (string.IsNullOrEmpty(instanceOwnerIdType) || string.IsNullOrEmpty(instanceOwnerIdValue))
+                    {
+                        return BadRequest("Invalid InstanceOwnerIdentifier.");
+                    }
+
+                    string orgNo = instanceOwnerIdType == "organization" ? instanceOwnerIdValue : string.Empty;
+                    string person = instanceOwnerIdType == "person" ? instanceOwnerIdValue : string.Empty;
+
+                    instanceOwnerPartyId = await _registerService.PartyLookup(orgNo, person);
```

Comment:
This value is actually never used. See the warning from Github below. The parameter values are being read twice. We do some basic validation here in the Controller, but the final values are taken from the original request. 

See line 193: `Dictionary<string, StringValues> queryParams = QueryHelpers.ParseQuery(Request.QueryString.Value);`

We might need to expand the _instanceRepository.GetInstancesFromQuery method with a new parameter for the partyId and update the method to ignore the queryParams value for party id.

Reply:
Alternative solution: https://github.com/Altinn/altinn-storage/pull/360/commits/fd47008e8351cc5e40cc82758abae33796ec5884